src/main/java/io/cdap/plugin/WindowAggregation.java

```diff
@@ -48,21 +65,62 @@
 @Description("Specify a window over which functions should be applied. \n" +
   "Supports functions: Rank, Dense Rank, Percent Rank, N tile, Row Number, Median," +
   "Continuous Percentile, Lead, Lag, First, Last, Cumulative distribution, Accumulate.")
-public class WindowAggregation extends SparkCompute<StructuredRecord, StructuredRecord> {
+public class WindowAggregation extends SparkCompute<StructuredRecord, StructuredRecord>
+  implements LinearRelationalTransform {
 
   public static final String NAME = "WindowAggregation";
+  // BigQuery specific aggregations
+  private static final Map<WindowAggregationConfig.Function, String> functionBQSqlMap =
+    new HashMap<WindowAggregationConfig.Function, String>() {{
+      put(WindowAggregationConfig.Function.RANK, "RANK()");
```

Comment:
Have you validated the generated SQL matches what Spark outputs for a window aggregation?

Reply:
Bump

Reply:
Yes